pkg/clusteragent/clusterchecks/dispatcher_test.go

```diff
@@ -423,7 +424,8 @@ func TestDanglingConfig(t *testing.T) {
 	assert.True(t, dispatcher.shouldDispatchDangling())
 
 	// get the danglings and make sure they are removed from the store
-	configs := dispatcher.retrieveAndClearDangling()
+	configs := dispatcher.retrieveDangling()
+	dispatcher.deleteDangling([]string{config.Digest()})
 	assert.Len(t, configs, 1)
 	assert.Equal(t, 0, len(dispatcher.store.danglingConfigs))
```

Comment:
Can we also test unscheduledChecks value in this test?

Reply:
Yes I recently discovered the test suite's `require.Eventually` method would be a perfect use case here to check after a certain period of time that the config is classified as unscheduledCheck. Adding this now